### PR TITLE
[DistDialect] Pir add mp precision alignment ut

### PR DESCRIPTION
**Pull request by hitywt**
### PR Category
Auto Parallel

### PR Types
New features

### Description
Pcard-67164

1. 基于PIR增加MP精度对齐的单测

### CODE REVIEW

File: paddle/fluid/pir/transforms/pd_op_to_kernel_pass.cc [hunk 1]
```diff
@@ -404,7 +404,7 @@ static pir::Value AddPlaceTransferOp(pir::Value in,
   } else if ((src_place.GetType() == phi::AllocationType::GPU ||
               src_place.GetType() == phi::AllocationType::XPU) &&
              (dst_place.GetType() == phi::AllocationType::CPU)) {
-    copy_kernel_key.set_backend(place2backend(dst_place.GetType()));
```
Comment (zhiqiu) [inline paddle/fluid/pir/transforms/pd_op_to_kernel_pass.cc]: if XPU, it should be dst_place.GetType()

File: test/auto_parallel/pir/semi_auto_parallel_dist_to_static_mlp_pir.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        )
+        dist_dataloader = dist.shard_dataloader(
+            dataloader=data_loader,
+            meshes=[self.mesh],
+        )
+        dy2static_losses, dist_model = self.run_dy2static(
+            dy2static_layer, dy2static_opt, dist_dataloader
+        )
+        dy_losses = self.run_dynamic(dy_layer, dy_opt, dist_dataloader)
+        np.testing.assert_allclose(dy_losses, dy2static_losses, rtol=1e-6)
```
Comment (zhiqiu) [inline test/auto_parallel/pir/semi_auto_parallel_dist_to_static_mlp_pir.py]: it should be equal

File: paddle/fluid/pir/transforms/pd_op_to_kernel_pass.cc (hunk 1 above)
Comment (hitywt) [reply paddle/fluid/pir/transforms/pd_op_to_kernel_pass.cc]: > if XPU, it should be dst_place.GetType()

I have confirmed with the XPU student that use src_place.GetType() is ok for gpu and xpu

Comment (zhiqiu) [reply paddle/fluid/pir/transforms/pd_op_to_kernel_pass.cc]: ok

File: python/paddle/distributed/auto_parallel/static/engine.py [hunk 3]
```diff
@@ -1022,6 +1023,17 @@ def _init_comm(self):
                 for process_group in all_process_groups:
                     process_group.instantiate()
 
+    def _init_lr(self):
+        buffer_tensor = global_scope().var("learning_rate_0").get_tensor()
```
Comment (JZ-LIANG) [inline python/paddle/distributed/auto_parallel/static/engine.py:1027]: 先判断 learning 是 float 设置，再向scope 创建var会更好。 might revise in future PR

**Review by JZ-LIANG — APPROVED** [review]
LGTM

**Review by zhangbo9674 — APPROVED** [review]

**Review by XieYunshen — APPROVED** [review]

**Pull request merged**